mi/instrument/kut/ek60/ooicore/driver.py

```diff
@@ -597,19 +601,38 @@ def echogram_thread(self):
                             continue
 
                         if result is not None:
-                            metadata, internal_timestamp = result
+                            metadata, internal_timestamp, data_times, power_data_dict, frequencies = result
 
                             filepath, timestamp = key
-                            log.info('Completed echogram with filepath: %r timestamp: %r', filepath, timestamp)
+                            log.info('Completed particles with filepath: %r timestamp: %r', filepath, timestamp)
 
-                            particle = ZplscBInstrumentDataParticle(metadata, port_timestamp=timestamp,
+                            metadata_particle = ZplscBInstrumentDataParticle(metadata, port_timestamp=timestamp,
                                                                     internal_timestamp=internal_timestamp,
                                                                     preferred_timestamp=DataParticleKey.INTERNAL_TIMESTAMP)
-                            parsed_sample = particle.generate()
+                            parsed_sample = metadata_particle.generate()
 
                             if self._driver_event:
                                 self._driver_event(DriverAsyncEvent.SAMPLE, parsed_sample)
 
+                            for counter, data_timestamp in enumerate(data_times):
+                                zp_data = {
+                                    ZplscBParticleKey.FREQ_CHAN_1: frequencies[1],
+                                    ZplscBParticleKey.VALS_CHAN_1: list(power_data_dict[1][counter]),
+                                    ZplscBParticleKey.FREQ_CHAN_2: frequencies[2],
+                                    ZplscBParticleKey.VALS_CHAN_2: list(power_data_dict[2][counter]),
+                                    ZplscBParticleKey.FREQ_CHAN_3: frequencies[3],
+                                    ZplscBParticleKey.VALS_CHAN_3: list(power_data_dict[3][counter]),
+                                }
+
+                                sample_particle = ZplscBSampleDataParticle(zp_data, port_timestamp=timestamp,
+                                                                    internal_timestamp=data_timestamp,
+                                                                    preferred_timestamp=DataParticleKey.INTERNAL_TIMESTAMP)
```

Comment:
For this cabled side, preferred timestamp is Internal timestamp because we will get timestamp for each slice of data.